SDWebImage/SDWebImageDownloaderOperation.m

```diff
@@ -403,6 +402,7 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didComp
              *  If you specified to use `NSURLCache`, then the response you get here is what you need.
              */
             __block NSData *imageData = [self.imageData copy];
+            self.imageData = nil;
```

Comment:
We can also reuse `self.imageData` directly, not to copy the data, but I don't like it, I want to keep data immutable.

Reply:
I think maybe the issue is because of the `__block` which will increate the lifetime of the data pointer.

This ugly design is because of that `decompressedImageWithImage:data`, will modify the original data pointer.

This feature (or hack) is removed in the 5.x, so we don't need that `__block`.

This fix seems OK, merge it.